library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -136,6 +123,20 @@ internal class TransactionPayloadFragment :
         super.onCreateOptionsMenu(menu, inflater)
     }
 
+    private fun shouldShowSaveIcon(transaction: HttpTransaction?) = when {
+        // SAF is not available on pre-Kit Kat so let's hide the icon.
+        (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT) -> false
+        (type == TYPE_REQUEST && 0L == (transaction?.requestContentLength ?: 0L)) -> false
+        (type == TYPE_RESPONSE && 0L == (transaction?.responseContentLength ?: 0L)) -> false
+        else -> true
+    }
+
+    private fun shouldShowSearchIcon(transaction: HttpTransaction?) = when {
```

Comment:
With my other suggestions comparisons seem too hard to read and understand. I suggest to rewrite the function the following way:
```
private fun shouldShowSearchIcon(transaction: HttpTransaction?) = when {
        (type == TYPE_REQUEST) -> {
            (true == transaction?.isRequestBodyPlainText) && (0L == (transaction.requestContentLength))
        }
        (type == TYPE_RESPONSE) -> {
            (true == transaction?.isResponseBodyPlainText) && (0L == (transaction.responseContentLength))
        }
        else -> false
    }
```